src/sys/wait.rs

```diff
@@ -260,3 +317,54 @@ pub fn waitpid<P: Into<Option<Pid>>>(pid: P, options: Option<WaitPidFlag>) -> Re
 pub fn wait() -> Result<WaitStatus> {
     waitpid(None, None)
 }
+
+/// The ID argument for `waitid`
+#[cfg(any(
+    target_os = "android",
+    target_os = "freebsd",
+    target_os = "haiku",
+    target_os = "linux",
+))]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum Id {
+    /// Wait for any child
+    All,
+    /// Wait for the child whose process ID matches the given PID
+    Pid(Pid),
+    /// Wait for the child whose process group ID matches the given PID
+    ///
+    /// If the PID is zero, the caller's process group is used since Linux 5.4.
+    PGid(Pid),
+    /// Wait for the child referred to by the given PID file descriptor
+    #[cfg(any(target_os = "android", target_os = "linux"))]
+    PIDFd(RawFd),
+}
+
+/// Wait for a process to change status
+///
+/// See also [waitid(2)](https://pubs.opengroup.org/onlinepubs/9699919799/functions/waitid.html)
+#[cfg(any(
```

Comment:
libc defines this function for almost all OSes.  I think it might be everything except Redox.  Any reason not to enable it for all of them in Nix?

Reply:
I started with the list of platforms for which `WEXITED` is defined in `WaitPidFlag` and then removed all that had some obvious issue:
- NetBSD has broken siginfo_t in libc (see issue linked in PR description)
- Apple/IOS are missing CLD_EXITED/KILLED/DUMPED in libc (and I don't really want to open *another* libc PR for a system that I do not use and that I'm not familiar with...)